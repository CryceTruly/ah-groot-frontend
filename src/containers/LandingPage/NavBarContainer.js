import React, { Component } from 'react';
import { connect } from 'react-redux';
import { storeCategories } from '../../actions';

export class NavBarContainer extends Component {
  componentWillMount() {
    this.props.storeCategories();
  }

  render() {
    return this.renderCategories();
  }

  renderCategories = () => {
    let categoryComponents = this.props.categories.slice(0, 6).map(element => {
      return (
        <li key={element.slug}>
          <a href="#">{element.name}</a>
        </li>
      );
    });
    if (this.props.categories.length > 6) {
      categoryComponents.push(
        <li key="more">
          <a href="#">More</a>
        </li>
      );
      return categoryComponents;
    }
    if (this.props.categories.length <= 6) {
    return categoryComponents;
    }
  };
}

const mapStateToProps = state => {
  return {
    articles: state.articles,
    categories: state.categories,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    storeCategories: function() {
      dispatch(storeCategories());
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavBarContainer);
