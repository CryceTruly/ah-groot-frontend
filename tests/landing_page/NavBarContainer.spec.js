import configureStore from 'redux-mock-store';
import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from "redux-thunk";
import { Provider } from 'react-redux';
import moxios from 'moxios';

import data from './maxios_mock';
import NavBarContainer from '../../src/containers/LandingPage/NavBarContainer';

describe('NavBarContainer', () => {
  const initialState = {
    categories :[{name:'Head lines', slug:'head-lines'}]
  };
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    moxios.install();
  });

  afterEach(function () {
    moxios.uninstall();
  });

  it('should render without crashing', () => {
    const wrapper = shallow( <Provider store={store}><NavBarContainer /></Provider> );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing 2', () => {
    const wrapper = shallow( <Provider store={store}><NavBarContainer /></Provider> );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when state is updated', () => {
    const wrapper = mount( <Provider store={store}><NavBarContainer /></Provider> );
    wrapper.setProps({ categories: data.category.categorys.results });
    expect(wrapper.prop('categories').length).toEqual(2);
  });
});

