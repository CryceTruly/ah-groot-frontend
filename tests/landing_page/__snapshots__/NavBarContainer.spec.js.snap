// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarContainer should render without crashing 1`] = `ShallowWrapper {}`;
